src/java/org/apache/cassandra/io/storage/StorageProvider.java

```diff
@@ -158,5 +206,53 @@ public void invalidateFileSystemCache(File file)
         {
             INativeLibrary.instance.trySkipCache(file, 0, 0);
         }
+
+        protected Config.DiskAccessMode accessMode(Component component)
+        {
+            switch (component.type)
+            {
+                case PRIMARY_INDEX:
+                case PARTITION_INDEX:
+                case ROW_INDEX:
```

Comment:
I think we need a  `case SECONDARY_INDEX` that returns mmap

Reply:
Those are SASI index and they currently don't use the `FileHandle` thing at all, they have their manual [`MappedBuffer`](https://github.com/datastax/cassandra/blob/ds-trunk/src/java/org/apache/cassandra/index/sasi/utils/MappedBuffer.java) with a custom API, and it doesn't seem easy to retrofit into `FileHandle`.

Should we care?

Reply:
my mistake, we don't care about sasi

i guess SAI is Custom instead (weird)

Reply:
From Slack
> SAI never use the method for Descriptor to create it's file handles. Instead it uses the other ones that take an IndexDescriptor , and that is hardcoded to mmap.